packages/two_dimensional_scrollables/lib/src/table_view/table_cell.dart

```diff
@@ -36,8 +36,147 @@ class TableVicinity extends ChildVicinity {
 }
 
 /// Parent data structure used by [RenderTableViewport].
+///
+/// The parent data primarily describes merged cell behavior for a child of the
+/// viewport. This is controlled by the [ParentDataWidget], [TableViewCell].
 class TableViewParentData extends TwoDimensionalViewportParentData {
+  /// The index of the first column the [TableViewCell] is merged with.
+  ///
+  /// Pinned cells cannot be merged with cells that are not pinned.
+  ///
+  /// Must be greater than or equal to zero. If [columnMergeEnd] is set, this
+  /// must not be null.
+  int? columnMergeStart;
+
+  /// The index of the last column the [TableViewCell] is merged with.
+  ///
+  /// Pinned cells cannot be merged with cells that are not pinned.
+  ///
+  /// If [columnMergeStart] is set, this must not be null.
+  int? columnMergeEnd;
```

Comment:
on a side note, may I suggest a different api that relies on column span ?
maintaining one index (columnMergeStart) + a length (columnSpan) 
is better than maintaining two indexes (columnMergeStart) + (columnMergeEnd)
since if the user wants to shift the merged cells, they only need to change the start index.

Reply:
Thanks for the feedback! That is a great idea. 
I am going to close this for now while we look into making some changes in ParentDataWidget so unblock the issue that cropped up here.